### DISCUSSION TITLE
SSO with GoogleProvider in headless browser using Cypress redirecting to different page

### DISCUSSION THREAD

**Discussion by AaronMcCloskey** [nextauthjs/next-auth#6333] [Help]
### Provider type

Google

### Environment

```
  System:
    OS: macOS 12.6
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 439.20 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.14.2/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
    Watchman: 2022.12.12.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 108.0.5359.124
    Edge: 95.0.1020.53
    Firefox: 108.0.1
    Safari: 15.6.1
  npmPackages:
    next: 13.0.2 => 13.0.2 
    next-auth: ^4.18.3 => 4.18.7 
    react: 18.2.0 => 18.2.0 
```

### Reproduction URL

N/A

### Describe the issue

I apologize if this is not actually related to `next-auth` in any way.

Using SSO with Google in my chrome browser I am brought to `https://accounts.google.com/signin/oath`, as does using Cypress Chrome browser, however when I run my Cypress tests in a headless chrome browser I get brought to `https://accounts.google.com/ServiceLogin`.

Which has caused me quite a headache trying to handle this possibility with it ending with me unable to actually click a button to allow access, so now a bit stuck.

Chrome in Cypress
![Screenshot 2023-01-09 at 04 27 18](https://user-images.githubusercontent.com/20436343/211241204-837e18db-128b-413b-94d2-9a5720e96bf1.png)

Headless Chrome is Cypress
<img width="826" alt="Screenshot 2023-01-09 at 04 26 05" src="https://user-images.githubusercontent.com/20436343/211241002-6ba1223e-f5aa-42d4-908d-0fc422c2089c.png">

### How to reproduce

Here is my `[...nextauth].ts` file. Do note, I tried to specify the `version` on the `GoogleProvider` options to be `2.0` but the issue still persists.

```
/* eslint-disable camelcase */
import { PrismaAdapter } from '@next-auth/prisma-adapter';
import type { User } from '@prisma/client';
import sgMail from '@sendgrid/mail';
import type { Session } from 'next-auth';
import NextAuth, { type NextAuthOptions } from 'next-auth';
import type { GoogleProfile } from 'next-auth/providers/google';
import GoogleProvider from 'next-auth/providers/google';
import { v4 } from 'uuid';

import { env } from '../../../env/server.mjs';
import { prisma } from '../../../server/db/client';
import { userRouter } from '../../../server/trpc/router/user';

export const authOptions: NextAuthOptions = {
  adapter: PrismaAdapter(prisma),
  providers: [
    GoogleProvider({
      clientId: env.GOOGLE_CLIENT_ID,
      clientSecret: env.GOOGLE_CLIENT_SECRET,
      version: '2.0',
      // For some reason we need this here for google login to work in Cypress?
      checks: 'none',
      async profile(p: GoogleProfile): Promise<User> {
        let emailToken: any = '';
        const user = await prisma.user.findUnique({ where: { email: p.email } });
        if (!user) emailToken = v4();
        return {
          id: p.sub,
          firstName: p.given_name,
          surname: p.family_name,
          email: p.email,
          image: p.picture,
          role: 'USER',
          emailVerified: null,
          createdAt: new Date(),
          updatedAt: new Date(),
          lastLoggedIn: new Date(),
          emailToken,
          token: null,
          unsubscribed: false,
        };
      },
    }),
  ],
  secret: env.NEXTAUTH_SECRET,
  session: {
    strategy: 'database',
    maxAge: 30 * 24 * 60 * 60,
    updateAge: 24 * 60 * 60,
  },
  jwt: {
    secret: env.NEXTAUTH_SECRET,
  },
  pages: {
    signIn: '/login',
    newUser: '/register',
  },
  events: {
    signIn: async ({ isNewUser, user }) => {
      if (isNewUser) {
        const session: Session = user;
        await userRouter.createCaller({ session, prisma, sgMail }).sendEmailVerification();
      } else {
        await prisma.user.update({ where: { id: user.id }, data: { lastLoggedIn: new Date() } });
      }
    },
  },
};

export default NextAuth(authOptions);
```

### Expected behavior

I'd expect that even on a headless browser that when SSO via Google to be redirected to `https://accounts.google.com/signin/oath`

**Comment by balazsorban44** [comment]
Testing OAuth providers in CI is tricky, most of them will have extra measures to filter out bots. I can imagine it's the case here too. I don't have an exact answer, but here are some community-written tutorials/tools that might help:
- https://next-auth.js.org/tutorials/testing-with-cypress
- https://github.com/TomFreudenberg/next-auth-mock by @TomFreudenberg who might also have extra thoughts
